packages/react-native-sdk/src/utils/push/android.ts

```
@@ -146,279 +129,254 @@ export const firebaseDataHandler = async (
+                callingx.endCallWithReason(callId, 'rejected');
+              }
+            },
+          );
+
           unsubscribeFunctions.push(unsubscribe);
-          unsubscribeFunctions.push(() => subscription.unsubscribe());
+          unsubscribeFunctions.push(() => stateSubscription.unsubscribe());
+          unsubscribeFunctions.push(() => endCallSubscription.remove());
           pushUnsubscriptionCallbacks.get(call_cid)?.forEach((cb) => cb());
           pushUnsubscriptionCallbacks.set(call_cid, unsubscribeFunctions);
         });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)